rmw_connext_dynamic_cpp/src/functions.cpp

```diff
@@ -1052,14 +1634,27 @@ rmw_create_subscription(const rmw_node_t * node,
           NULL, \
           i + 1); \
         if (status != DDS_RETCODE_OK) { \
-          delete[] values; \
+          if (values) { \
+            /* First call the destructor for each element, then free the whole block. */ \
+            for (size_t i = 0; i < array_size; ++i) { \
+              RMW_TRY_DESTRUCTOR(values[i].~DDS_TYPE(), DDS_TYPE, ) \
+            } \
+            rmw_free(values); \
+          } \
```

Comment:
Since this block is repeated numerous time should it be moved into a separate macro?